### ISSUE TITLE
Submodules in detached state

### ISSUE THREAD

**Issue by ahood** [jamesturk/jellyfish#112]
When I forked this repository and cloned to my laptop, the cjellyfish and testdata submodules were in a detached state. Each of them did have master branches, however, where HEAD and master were equal for cjellyfish and master was ahead of HEAD for testdata. This is my first time seeing submodules. Is something messed up that needs to be fixed? I'm planning to just switch to master in both cases before I do a pull request. Is that the right thing to do?

**Comment by ahood** [comment]
In order to pass checks in my pull request I had to change .gitmodules to point to the submodules in my fork, following what another user did in a separate pending PR. The checks are now passing. Is this the right way to deal with forking and pull requests for a repository with submodules? Perhaps you would consider adding instructions for how submodule newbies like myself should interact with the jellyfish repo and its submodules.

I found this reddit post helpful: https://www.reddit.com/r/git/comments/qatxi/fork_a_github_repo_that_has_submodule/.
